ScreensPanel.c

```diff
@@ -132,16 +132,25 @@ static void startRenaming(Panel* super) {
    Panel_setCursorToSelection(super);
 }
 
-static void rebuildSettingsArray(Panel* super) {
+static void rebuildSettingsArray(Panel* super, int selected) {
    ScreensPanel* const this = (ScreensPanel*) super;
 
    int n = Panel_size(super);
    free(this->settings->screens);
    this->settings->screens = xMallocArray(n + 1, sizeof(ScreenSettings*));
    this->settings->screens[n] = NULL;
+   bool resetScreenIndex = true;
    for (int i = 0; i < n; i++) {
       ScreenListItem* item = (ScreenListItem*) Panel_get(super, i);
       this->settings->screens[i] = item->ss;
+      if (item->ss == this->settings->ss)
+         resetScreenIndex = false;
```

Comment:
Is there any necessaty to skip doing the re-assignment when these pointers match? Won't hurt AFAICS. I'd opt to skip the `resetScreenIndex` check ensuring that even in cases we don't anticipate the below re-assignment re-establishes a clean state.

Reply:
@BenBE yep, I think you're right.  I've updated the code to behave this way instead and tested its all OK still - looks good.